eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/ChannelOpenDualFunded.scala

```diff
@@ -362,23 +363,156 @@ trait ChannelOpenDualFunded extends DualFundingHandlers with ErrorHandlers {
             stay() using d1 storing() calling publishFundingTx(d.fundingParams, fundingTx)
         }
         case _: FullySignedSharedTransaction =>
-          log.warning("received duplicate tx_signatures")
-          stay()
+          d.rbfAttempt match {
+            case Some(Right(txBuilder)) =>
+              txBuilder ! InteractiveTxBuilder.ReceiveTxSigs(txSigs)
+              stay()
+            case _ =>
+              // Signatures are retransmitted on reconnection, but we may have already received them.
+              log.info("ignoring duplicate tx_signatures for txid={}", txSigs.txId)
+              stay()
+          }
       }
 
-    case Event(_: TxInitRbf, _: DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED) =>
-      log.info("rbf not supported yet")
-      stay()
+    case Event(msg: TxInitRbf, d: DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED) =>
+      if (d.fundingParams.isInitiator) {
+        // Only the initiator is allowed to initiate RBF.
+        log.info("rejecting tx_init_rbf, we're the initiator, not them!")
+        stay() sending TxAbort(d.channelId, InvalidRbfAttempt(d.channelId).getMessage)
```

Comment:
Shouldn't we return an `Error` instead? This is a protocol violation after all.

Reply:
There are two other cases in that handler.

Reply:
Actually I reverted that to a `TxAbort`  in https://github.com/ACINQ/eclair/pull/2275/commits/135870bfb2f1660d74ab3eac7a07475b121651bb for the following reason: sending an `error` here means we'll close the channel, which may cost us on-chain fees. We have a valid funding tx waiting to be confirmed, so it's a bit of a waste to force-close because of an rbf issue that we can safely ignore?

Reply:
If we refer to the spec, `TxAbort` isn't related to errors:
> The `tx_abort` message allows for the cancellation of an in progress negotiation, and a return to the initial starting state.

So I would argue that we could instead send a `warning`, if we don't want the channel to close. But we're nitpicking, reverting to `TxAbort` is fine.

Reply:
The benefit of sending `TxAbort` is that it puts the channel back into a valid state: they did an invalid RBF operation, so we just abort the rbf attempt, and the channel is in a clean state again, does that make sense?

We could send a warning on top of that, but since `TxAbort` already includes a failure message, I believe it would be redundant?

Reply:
Let's leave it that way and see how cln implements?